src/SingleSignOnDemo/SSOExample.cs

```diff
@@ -0,0 +1,129 @@
+
+                // Ein Box-Object erzeugen, die zur Kommunikation mit mySMS unter Verwendung des öffentlichen und privaten Schlüssels dient.
+                // Create a Box object to communicate with mySMS using the public and private key.
+                // Créer un object Box pour communiquer avec mySMS en utilisant les clés publique et privée.
+                byte[] encrypted = PublicKeyBox.Create(buffer, nonce, Convert.FromBase64String(APPLICATION_PRIVATE_KEY), Convert.FromBase64String(PLATFORM_PUBLIC_KEY));
+
+                // Nonce und verschlüsselten Text aneinanderreihen und in einen Base64-String umwandeln.
+                // Concatenate the nonce and the encrypted text and convert it to a base64 string.
+                // Concaténer le nonce et le texte chiffré et convertisser-le en chaîne base64.
+                byte[] payload = nonce.ToList().Concat(encrypted.ToList()).ToArray();
+                string base64 = Convert.ToBase64String(payload, Base64FormattingOptions.None);
+
+                // Eine HTTP-Anfrage an die Plattform senden.
+                // Send an HTTP request to the platform.
+                // Envoyer une requête HTTP à la plate-forme.
+                string sso_url = string.Format("{0}/auth/lookup/{1}", SSO_BASE_URL, token);
+                HttpWebRequest req = WebRequest.CreateHttp(sso_url);
+                req.Headers.Add("Authorization", string.Format("PRODUCTAUTH {0}:{1}", PRODUCT_NAME, base64));
+                req.Method = "GET";
+
+                // Die Antwort empfangen und verarbeiten
+                // Receive and proceed the response.
+                // Recever et passer la réponse.
+                HttpWebResponse resp = (HttpWebResponse)req.GetResponse();
+                if (resp.StatusCode == HttpStatusCode.OK)
+                {
+                    using (MemoryStream ms = new MemoryStream())
+                    {
+                        resp.GetResponseStream().CopyTo(ms);
+                        try
+                        {
+                            // Die Antwort enthält die 24 Byte lange Nonce und die verschlüsselten Benutzerdaten.
+                            // The response contains the nonce with a size of 24 bytes and the encrypted user data.
+                            // La réponse contient la nonce avec un longueur de 24 octets et les données d'utilisateur cryptées.
+                            byte[] encr_nonce = ms.ToArray().Take(24).ToArray();
+                            byte[] encr_data = ms.ToArray().Skip(24).ToArray();
+                            byte[] decrypted = PublicKeyBox.Open(encr_data, encr_nonce, Convert.FromBase64String(APPLICATION_PRIVATE_KEY), Convert.FromBase64String(PLATFORM_PUBLIC_KEY));
+                        }
+                        catch (CryptographicException ex)
+                        {
+                            Console.WriteLine(ex.Message);
+                            return false;
```

Comment:
This feels redundant to the outer exception. Not sure about best practices in .Net tough :)